### PR TITLE
[Network] Convert Load Balancer and App Gateway Create to Dynamic Templates

### LINKED ISSUE #2198: [Network] Remove dependencies on external storage account
In order to support Profiles, the following create commands need to be converted to use either SDK calls or dynamic template generation.

**Template Based**
- [x] Application Gateway (PR #2668)
- [x] Load Balancer (PR #2668)

**SDK Based**
- [x] Express Route Circuit (PR #2371)
- [x] Local Gateway (PR #2371)
- [x] NIC (PR #2371)
- [x] NSG (PR #2371)
- [x] Public IP (PR #2294)
- [x] Traffic Manager Profile (PR #2371)
- [x] VNET (PR #2269)
- [x] VNET Gateway (PR #2371)

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #2198.

### CODE REVIEW

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 1]
```diff
@@ -130,29 +130,37 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
...
+register_cli_argument('network application-gateway', 'servers', nargs='+', help='Space separated list of IP addresses or DNS names corresponding to backend servers.', validator=validate_servers, arg_group='Gateway')
+register_cli_argument('network application-gateway', 'http_settings_cookie_based_affinity', cookie_based_affinity_type, help='Enable or disable HTTP settings cookie-based affinity.', arg_group='Gateway')
+register_cli_argument('network application-gateway', 'http_settings_protocol', http_protocol_type, help='The HTTP settings protocol.', arg_group='Gateway')
+register_cli_argument('network application-gateway', 'subnet_address_prefix', help='The CIDR prefix to use when creating a new subnet.', action=MarkSpecifiedAction, arg_group='Network')
+register_cli_argument('network application-gateway', 'vnet_address_prefix', help='The CIDR prefix to use when creating a new VNet.', action=MarkSpecifiedAction, arg_group='Network')
+register_cli_argument('network application-gateway', 'virtual_network_type', ignore_type, arg_group='Network')
+register_cli_argument('network application-gateway', 'private_ip_address_allocation', ignore_type, arg_group='Network')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: This appears a dupe with the same declaration 8 lines above

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py [hunk 2]
```diff
@@ -423,24 +431,30 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
...
-register_cli_argument('network lb create', 'private_ip_address_allocation', ignore_type)
-register_cli_argument('network lb create', 'dns_name_type', ignore_type)
+for item in ['backend_pool_name', 'backend_address_pool_name']:
+    register_cli_argument('network lb', item, options_list=('--backend-pool-name',), help='The name of the backend address pool.', completer=get_lb_subresource_completion_list('backend_address_pools'))
+
+register_cli_argument('network lb create', 'validate', help='Generate and validate the ARM template without creating any resources.', action='store_true', validator=process_lb_create_namespace)
+
+register_cli_argument('network lb create', 'public_ip_dns_name')
+register_cli_argument('network lb create', 'public_ip_address_allocation', **enum_choice_list(IPAllocationMethod))
+register_cli_argument('network lb create', 'public_ip_dns_name', help='Globally unique DNS name for a new public IP.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py]: dupe with the same one 2 lines above?

File: src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py [hunk 3]
```diff
@@ -92,17 +91,99 @@ def list_application_gateways(resource_group_name=None):
...
+
+    template = master_template.build()
+
+    # deploy ARM template
+    deployment_name = 'ag_deploy_' + random_string(32)
+    client = get_mgmt_service_client(ResourceManagementClient).deployments
+    properties = DeploymentProperties(template=template, parameters={}, mode='incremental')
+    if validate:
+        from pprint import pprint
+        pprint(template)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py]: you can consider to reuse https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py#L49

**Comment by tjprescott** [comment]
@yugangw-msft comments addressed.

**Pull request merged**